src/lib/operators/insert.cpp

```diff
@@ -97,6 +97,10 @@ std::shared_ptr<const Table> Insert::_on_execute(std::shared_ptr<TransactionCont
    *    Do so while locking the table to prevent multiple threads modifying the table's size simultaneously.
    *    Since allocation is expected to be faster than writing to the memory, allocating under lock and then writing -
    *    in a second step - without lock will minimize the time that the Table's append_mutex is locked.
+   *
+   *    In the rare case where the input table is the table that we write to (OperatorsInsertTest.SelfInsert), this
```

Comment:
I am not sure such vague comments pointing in the direction of possible complications if a problem would have been attempted to be solved in a different manner are very helpful...